internal/pkg/api/handleCheckin.go

```diff
@@ -119,6 +119,11 @@ func (ct *CheckinT) handleCheckin(zlog zerolog.Logger, w http.ResponseWriter, r
 
 	agent, err := authAgent(r, &id, ct.bulker, ct.cache)
 	if err != nil {
+		// invalidate remote API keys of force unenrolled agents
+		if err == ErrAgentInactive && agent != nil {
```

Comment:
Added testing steps of the force unenroll scenario to the pr description.

Decided to invalidate remote API keys on checkin here, when finding that the agent is inactive, because the agent becomes inactive after force unenroll. It's not guaranteed that the agent is going to check-in after force unenroll, but we can't do the invalidation in kibana, as kibana doesn't have access to the remote es service token as it is a secret. 
Fleet-server also doesn't handle force unenroll actions, and even if it would, fleet server only reads actions on agent checkin. So I didn't find a better way to invalidate api keys after force unenroll.